service/src/main/java/uk/nhs/adaptors/gp2gp/common/task/TaskConsumer.java

```diff
@@ -0,0 +1,38 @@
+package uk.nhs.adaptors.gp2gp.common.task;
+
+import java.io.IOException;
+
+import javax.jms.JMSException;
+import javax.jms.Message;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jms.annotation.JmsListener;
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import uk.nhs.adaptors.gp2gp.common.exception.TaskHandlerException;
+
+@Component
+@Slf4j
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class TaskConsumer {
+
+    private final TaskHandler taskService;
+
+    @JmsListener(destination = "${gp2gp.amqp.taskQueueName}")
```

Comment:
`gp2gp.amqp.taskQueueName` should be configurable in application.yml using an environment variable. This environment variable must be documented in README.md. It should have default value of "gp2gpTaskQueue".